Rematch/UpdateDB.lua

```diff
@@ -0,0 +1,35 @@
+local ns = select(2, ...)
+local Addon = ns.Addon
+
+function Addon.ResetEverything()
+    local scriptsDB = PetBattleScripts.db.global.scripts
+
+    scriptsDB.Rematch = CopyTable(scriptsDB.Rematch4)
+    wipe(scriptsDB.Rematch4)
+end
+
+function Addon:UpdateDB(convertedTeams)
+    -- Backup old scripts
+    local scriptsDB = PetBattleScripts.db.global.scripts
+    scriptsDB.Rematch4 = CopyTable(scriptsDB.Rematch)
+
+    -- First we need a cache list of all of our scripts.
+    -- so we can modify our scripts database without messing up the loops
+    local scriptList = {}
+    for key, script in self:IterateScripts() do
+        scriptList[key] = script
+    end
+
+    -- now we can migrade scripts.
+    for key, script in pairs(scriptList) do
+        local newTeamID = convertedTeams[key]
+        if newTeamID then
+            self:RemoveScript(key)
+            self:AddScript(newTeamID, script)
+
+            --TODO: do we want to maintain the same script name and team name?
```

Comment:
open todo

> do we want to maintain the same script name and team name?

Reply:
In `UpdateDB.lua` I'm already doing that on migration. I think doing it here would be ok.

Reply:
Oh whoops this comment is in the `UpdateDB.lua` file. That sync is already happening in the `REMATCH_TEAM_UPDATED` event, so i don't see why it shouldn't be here too.